### PR TITLE
fix: Backport limit spin loop in TLSActor #29922

### PR DESCRIPTION
**Pull request by johanandren**
Refs #29922 

Backport of fix from 2.8 release line.

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala [hunk 1]
```diff
@@ -480,7 +492,6 @@ import akka.util.ByteString
     pump()
   }
 
-  // FIXME: what happens if this actor dies unexpectedly?
```
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala:483]: Accidental cherry pick removal but doesn't really matter.

**Review by octonato — APPROVED** [review]
LGTM

**Comment by octonato** [comment]
Oh, it seems that branch is still configured with Travis which won't run and GH Actions didn't kick in either.

File: akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala [hunk 2]
```diff
@@ -408,7 +410,17 @@ import akka.util.ByteString
...
+            unwrapPutBackCounter += 1
+            if (unwrapPutBackCounter > 1000) {
+              throw new IllegalStateException(
+                s"Stuck in unwrap loop, bailing out, last handshake status [$lastHandshakeStatus], " +
+                s"remaining=${transportInBuffer.remaining}, out=${userOutBuffer.position()}, " +
+                "(https://github.com/akka/akka/issues/29922)")
+            }
             transportInChoppingBlock.putBack(transportInBuffer)
           case FINISHED =>
             flushToUser()
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala:426]: Comparing main and 2.6 I see that there is another change that might be related but I don't know. https://github.com/akka/akka/pull/31433/files

Comment (johanandren) [reply akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala:426]: Yeah, wasn't sure that was related, so opted to not backport that when I was looking at it. Not sure that is the right decision though, that flush-change is also pretty minimal.

**Comment by johanandren** [comment]
close-open to trigger CI

**Comment by johanandren** [comment]
Meh, old versions of github actions and the security settings for those blocking CI from running now.

**Pull request merged**